bin/git-co-upstream-pr

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+pull_request_number=$1
+local_branch_name=$2
+
+if git remote -v | grep upstream; then
+  git fetch upstream "pull/$pull_request_number/head:$local_branch_name"
+  git checkout "$local_branch_name"
+else
+  echo "You don't have an upstream remote set.
```

Comment:
Up to you, but two of the things you're doing are unreliable (but not necessarily in an important way):

- `echo` is considered fragile unless you pass very, very boring things to it. You're passing a boring thing, but it could be more boring. Makes me nervous but it's probably fine.
- Multi-line quotes have a fraught history. Again, it will probably work on modern computers, but it makes me nervous.

Here's another idea:

```sh
cat <<HELP
You don't have an upstream remote set.
Use:
  git remote add upstream {upstream_remote_url}
to set the reference and then try again.
HELP
```

This uses a heredoc (`here-doc`, according to one manpage I read), which has more support across shells, and sends the output of that straight into `cat`.

Reply:
Oh, I didn't know that! I was following the existing format from the other commands. I do not want to do unreliable things :O

Thank you for this lesson and suggestion, Mike!

Changed this in d2069f516a62915b6f7e4c2d84dbbcee096247f3